modin/core/dataframe/pandas/exchange/dataframe_protocol/from_dataframe.py

```diff
@@ -248,38 +278,49 @@ def datetime_column_to_ndarray(col: ProtocolColumn) -> Tuple[np.ndarray, Any]:
     dbuf, dtype = buffers["data"]
     # Consider dtype being `uint` to get number of units passed since the 01.01.1970
     data = buffer_to_ndarray(
-        dbuf, (DTypeKind.UINT, dtype[1], "u", "="), col.offset, col.size
+        dbuf,
+        (
+            DTypeKind.UINT,
+            dtype[1],
+            getattr(ArrowCTypes, f"UINT{dtype[1]}"),
+            Edianness.NATIVE,
+        ),
+        col.offset,
+        col.size,
     )
 
-    if format_str.startswith("ts"):
+    def parse_format_str(format_str, data):
+        """Parse datetime `format_str` to interpret the `data`."""
         # timestamp 'ts{unit}:tz'
-        meta = format_str[2:].split(":")
-        if len(meta) == 1:
-            unit = meta[0]
-            tz = ""
-        else:
-            unit, tz = meta
-        if tz != "":
-            raise NotImplementedError("Timezones are not supported yet")
-        if unit != "s":
-            # the format string describes only a first letter of the unit, add one extra
-            # letter to make the unit in numpy-style: 'm' -> 'ms', 'u' -> 'us', 'n' -> 'ns'
-            unit += "s"
-        data = data.astype(f"datetime64[{unit}]")
-    elif format_str.startswith("td"):
+        timestamp_meta = re.findall(r"ts([smun]):(.*)", format_str)
+        if timestamp_meta:
+            unit, tz = timestamp_meta[0]
+            if tz != "":
+                raise NotImplementedError("Timezones are not supported yet")
+            if unit != "s":
+                # the format string describes only a first letter of the unit, add one extra
+                # letter to make the unit in numpy-style: 'm' -> 'ms', 'u' -> 'us', 'n' -> 'ns'
+                unit += "s"
+            data = data.astype(f"datetime64[{unit}]")
+            return data
+
         # date 'td{Days/Ms}'
-        unit = format_str[2:]
-        if unit == "D":
-            # NumPy doesn't support DAY unit, so converting days to seconds
-            # (converting to uint64 to avoid overflow)
-            data = (data.astype(np.uint64) * (24 * 60 * 60)).astype("datetime64[s]")
-        elif unit == "m":
-            data = data.astype("datetime64[ms]")
-        else:
-            raise NotImplementedError(f"Date unit is not supported: {unit}")
-    else:
+        date_meta = re.findall(r"td([Dm])", format_str)
```

Comment:
again, why `.findall()`? what if `format_str = "aaa tdM"`?.. this shouldn't be considered valid